third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv2.cpp

```diff
@@ -707,18 +788,161 @@ LogicalResult convertDot(const LLVMTypeConverter *typeConverter,
   return success();
 }
 
-// Convert to mma.m16n8k?
 LogicalResult convertMMA(triton::DotOp op, triton::DotOp::Adaptor adaptor,
                          const LLVMTypeConverter *typeConverter,
                          ConversionPatternRewriter &rewriter, bool isTuring,
                          bool isHopperF64) {
-  assert(mlir::isa<DotOperandEncodingAttr>(op.getA().getType().getEncoding()) &&
-         mlir::isa<DotOperandEncodingAttr>(op.getB().getType().getEncoding()) &&
-         "Both $a and %b should be DotOperand layout.");
+  auto aTensorTy = cast<RankedTensorType>(op.getA().getType());
+  auto bTensorTy = cast<RankedTensorType>(op.getB().getType());
+  auto dTensorTy = cast<RankedTensorType>(op.getD().getType());
```

Comment:
nit: are those cast really needed? getType should return a RankedTensorType already?

Reply:
removed.